internal/armada/server/submit.go

```diff
@@ -897,6 +897,22 @@ func (server *SubmitServer) applyDefaultsToPodSpec(spec *v1.PodSpec) {
 	if shouldDefaultPriorityClass {
 		spec.PriorityClassName = defaultPriorityClass
 	}
+
+	// add missing TerminationGracePeriod if needed
+	server.applyTerminationGracePeriodDefault(spec)
+}
+
+// applyTerminationGracePeriodDefault will give the podspec a default if needed
+func (server *SubmitServer) applyTerminationGracePeriodDefault(spec *v1.PodSpec) {
+	specNeedsTerminationGracePeriod := spec.TerminationGracePeriodSeconds == nil
+	defaultTerminationGracePeriod := int64(server.schedulingConfig.DefaultTerminationGracePeriod.Seconds())
+	defaultTerminationGracePeriodAvailable := defaultTerminationGracePeriod > 0
```

Comment:
I don't think this is right. We should allow setting the default to be 0s, which this check prevents.  I'd make this function do the following and nothing else:
- Check if `spec.TerminationGracePeriodSeconds` is `nil`.
- If it is, set it to `server.schedulingConfig.DefaultTerminationGracePeriod.Seconds()`.